libs/sql-schema-describer/src/mssql.rs

```diff
@@ -0,0 +1,570 @@
+
+    async fn get_size(&self, schema: &str) -> usize {
+        debug!("Getting db size");
+
+        let sql = r#"
+            SELECT SUM(alloUni.used_pages) * 8000 AS size
+            FROM sys.tables sysTab
+            INNER JOIN sys.indexes ind
+            ON sysTab.OBJECT_ID = ind.OBJECT_ID
+            AND ind.Index_ID <= 1
+            INNER JOIN sys.partitions parti
+                ON ind.OBJECT_ID = parti.OBJECT_ID
+                AND ind.index_id = parti.index_id
+            INNER JOIN sys.allocation_units alloUni
+                ON parti.partition_id = alloUni.container_id
+            WHERE SCHEMA_NAME(sysTab.SCHEMA_ID) == @P1
+            AND sysTab.is_ms_shipped = 0
+            AND ind.OBJECT_ID > 255
+            AND parti.rows > 0
+        "#;
+
+        let rows = self.conn.query_raw(sql, &[schema.into()]).await.expect("get db size");
+
+        let size: i64 = rows
+            .into_single()
+            .map(|row| row.get("size").and_then(|x| x.as_i64()).unwrap_or(0))
+            .unwrap_or(0);
+
+        size.try_into().unwrap()
+    }
+
+    fn get_table(
+        &self,
+        name: &str,
+        columns: &mut HashMap<String, Vec<Column>>,
+        indexes: &mut HashMap<String, (BTreeMap<String, Index>, Option<PrimaryKey>)>,
+        foreign_keys: &mut HashMap<String, Vec<ForeignKey>>,
+    ) -> Table {
+        let columns = columns.remove(name).expect("table columns not found");
+        let (indices, primary_key) = indexes.remove(name).unwrap_or_else(|| (BTreeMap::new(), None));
+
+        let foreign_keys = foreign_keys.remove(name).unwrap_or_default();
+
+        Table {
+            name: name.to_string(),
+            columns,
+            foreign_keys,
+            indices: indices.into_iter().map(|(_k, v)| v).collect(),
+            primary_key,
+        }
+    }
+
+    async fn get_all_columns(&self, schema: &str) -> HashMap<String, Vec<Column>> {
+        let sql = r#"
+            SELECT
+                column_name,
+                data_type,
+                character_maximum_length,
+                column_default,
+                is_nullable,
+                columnproperty(object_id(@P1 + '.' + table_name), column_name, 'IsIdentity') is_identity,
+                table_name
+            FROM information_schema.columns
+            WHERE table_schema = @P1
+            ORDER BY ordinal_position;
+        "#;
+
+        let mut map = HashMap::new();
+
+        let rows = self
+            .conn
+            .query_raw(sql, &[schema.into()])
+            .await
+            .expect("querying for columns");
+
+        for col in rows {
+            debug!("Got column: {:?}", col);
+
+            let table_name = col
+                .get("table_name")
+                .and_then(|x| x.to_string())
+                .expect("get table name");
+
+            let name = col
+                .get("column_name")
+                .and_then(|x| x.to_string())
+                .expect("get column name");
```

Comment:
Nitpick: the `serde` feature in quaint works pretty well for that kind of boilerplate.

Reply:
Yeah. We should use it here actually. Just wanted to follow the style of the describer for now, let's refactor later!